File: TD4/app/src/main/java/com/example/newslist/LoginActivity.java
package com.example.newslist;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class LoginActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        setTitle(getLocalClassName());

        final Button news_Btn = (Button) findViewById(R.id.login_button);

        news_Btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(LoginActivity.this, NewsActivity.class);
                startActivity(intent);
            }
        });
    }

    @Override
    public void onBackPressed() {
        finish();
    }
}

File: TD4 - Copie/app/src/main/java/com/example/newslist/LoginActivity.kt
package com.example.newslist

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity

class LoginActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        title = localClassName
        val news_Btn = findViewById<View>(R.id.login_button) as Button
        val username = findViewById<View>(R.id.username_edittext) as EditText
        val app = applicationContext as NewsListApplication
        val login = app.login
        news_Btn.setOnClickListener {
            val intent = Intent(this@LoginActivity, NewsActivity::class.java)
            intent.putExtra("login", username.text.toString())
            app.login = username.text.toString()
            startActivity(intent)
        }
    }

    override fun onBackPressed() {
        finish()
    }
}